### ISSUE TITLE
Problem: query block json-rpc api occasionally fail

### ISSUE THREAD

**Issue by yihuang** [evmos/ethermint#1269]
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]

__Steps to reproduce:__

query block occasionally fail:
```
'code': -32000, 'message': 'rpc error: code = Unknown desc = rpc error: code = Internal desc = failed to obtain coinbase address: failed to retrieve validator from block proposer address tcrcvalcons1v0rpejs3vanaccqrx4xkez3mlkmpxnugj3ga55: validator does not exist: unknown request'
```

__Expected behavior:__ should always work

__Actual behavior:__ [What actually happened]

__Additional info:__ [Include gist of relevant config, logs, etc.]

**Comment by VictorTrustyDev** [comment]
I am facing this issue also and have some more information about it
Let's say about [this DIFFUSION contract 0x3f75ceabCDfed1aCa03257Dc6Bdc0408E2b4b026](https://evm.evmos.org/address/0x3f75ceabCDfed1aCa03257Dc6Bdc0408E2b4b026), it was [created at block 
59500](https://evm.evmos.org/tx/0x742cabc183cb99980c898998d98c489c2d11cf89c8301df8203f87eaa5e261c6) so I query the contract name with context height 59500+1=59501, I got `rpc error: code = Unknown desc = rpc error: code = Internal desc = failed to obtain coinbase address: failed to retrieve validator from block proposer address evmosvalcons1z0tmsqqlt725lmsxjhn2sm09jkj6zcvv4jze8s: validator does not exist: unknown request`
Weird is this is the [NOMIC.ONE validator since block 2673107](https://escan.live/validator/evmosvalcons1z0tmsqqlt725lmsxjhn2sm09jkj6zcvv4jze8s) and this validator was created at [2672608 by this transaction](https://evmos.bigdipper.live/transactions/1BE6FB001E088C15F779E0CA47F3C1653BBA687D5520050A0755829E17025F73) (different name but same valoper) which is definately far away from 59501 and after I try again, ethclient.ContractCall still returns same error but the proposer became `evmosvalcons1nsczfx3qr75f3anp4lklcanm585x7vwfuw3mt4` is [Hanchon.live since block 486371](https://escan.live/validator/evmosvalcons1nsczfx3qr75f3anp4lklcanm585x7vwfuw3mt4), created at [461420](https://evmos.bigdipper.live/transactions/071C29F623BD1997D8BC2638EAF879A5AB3BB0B87D0329129462E0A974D7408E)

This is how I get contract name
```go
slicedEvmAddr := evmAddr[2:]
contract := common.HexToAddress(evmAddr)
callMsg := ethereum.CallMsg{
	To:   &contract,
	Data: common.FromHex(fmt.Sprintf("%s000000000000000000000000%s", funcSelector, slicedEvmAddr)),
}
ctxHeightBi := big.NewInt(height + 1)
queryCtx := rpctypes.ContextWithHeight(ctxHeightBi.Int64())
resC, errC = cp.ethClient.CallContract(queryCtx, callMsg, ctxHeightBi)
```
(I already query successfully many contracts)

Let's refer to this [error in ethermint project](https://github.com/evmos/ethermint/blob/c729467d112fb2ff6976addd77dfe406f6cbac56/x/evm/keeper/state_transition.go#L48) which is "failed to obtain coinbase address" and [this code (ethermint also)](https://github.com/evmos/ethermint/blob/c729467d112fb2ff6976addd77dfe406f6cbac56/x/evm/keeper/state_transition.go#L521) we see the proposer from BlockHeader (proto Header)
At first, I guess somehow the BlockHeader mistakenly becomes "current latest block" but there is another contract get name failed with same error but [this validator](https://escan.live/validator/evmosvaloper1n4zajqylnmk7kmnymq4ceynzlkq68j9l6qrxmg) which was stopped working for 2 days and I just query it today

Anyone have any idea?

**Comment by VictorTrustyDev** [comment]
> I think the [Proposer in this code](https://github.com/evmos/ethermint/blob/c729467d112fb2ff6976addd77dfe406f6cbac56/x/evm/keeper/state_transition.go#L521) somehow is Current proposer of Current state on the node

It's totally correct. This is how I debug it:
- I have 2 servers, one to query data, one is node, ensure time on both servers are the same
- I print debug message when I query data

> {"level":"debug","height":59140,"time":"2022-09-09T15:29:10Z","message":"NAME 0xd4949664cd82660aae99bedc034a0dea8a0bd517"}
> {"level":"debug","time":"2022-09-09T15:29:40Z","message":"FINISH name 0xd4949664cd82660aae99bedc034a0dea8a0bd517"}
> {"level":"error","contract":"0xd4949664cd82660aae99bedc034a0dea8a0bd517","error":"rpc error: code = Unknown desc = rpc error: code = Internal desc = failed to obtain coinbase address: failed to retrieve validator from block proposer address evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd: validator does not exist: unknown request","time":"2022-09-09T15:29:40Z","message":"failed to get contract name"}

We have err point out this time the proposer is [evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd, that is "Smart Stake | evmos.smartstake.io" since block 873482](https://escan.live/validator/evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd) (not exists at request height = 59141)

And now this is log on my node:

> 3:29PM INF indexed block height=3775696 module=txindex server=node
> 3:29PM INF Served eth_call conn=my.ip:port duration=73.683148 err="rpc error: code = Unknown desc = rpc error: code = Internal desc = failed to obtain coinbase address: failed to retrieve validator from block proposer address evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd: validator does not exist: unknown request" module=geth reqid=53
> 3:29PM INF Served eth_call conn=my.ip:port duration=1360.187138 err="rpc error: code = Unknown desc = rpc error: code = Internal desc = failed to obtain coinbase address: failed to retrieve validator from block proposer address evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd: validator does not exist: unknown request" module=geth reqid=54
> 3:29PM INF Transactions per second tps=0.7
> 3:29PM INF executed block height=3775697 module=state num_invalid_txs=0 num_valid_txs=2 server=node

We see node indexed and say **block 3775696** before the error message, and then node says **INF executed block height=3775697** after the error message. You know what? [evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd aka Smart Stake | evmos.smartstake.io is the proposer of 3775696](https://escan.live/validator/evmosvalcons10583wftddv3vqnn7cuqrsd84hj9r4u2yhjg5nd)

**Comment by yihuang** [comment]
should be fixed by #1391
